cms/datavis/blocks/base.py

```diff
@@ -0,0 +1,201 @@
+from collections.abc import Sequence
+from typing import Any, ClassVar, Optional
+
+from django.forms.widgets import Media
+from django.utils.functional import cached_property
+from wagtail import blocks
+from wagtail.blocks.struct_block import StructValue
+
+from cms.datavis.blocks.table import SimpleTableBlock
+
+
+class PointAnnotationBlock(blocks.StructBlock):
+    label = blocks.CharBlock(required=True)
+    x_position = blocks.IntegerBlock(label="x-position", required=True)
+    y_position = blocks.IntegerBlock(label="y-position", required=True)
+
+
+class BaseVisualisationBlock(blocks.StructBlock):
+    title = blocks.CharBlock()
+    subtitle = blocks.CharBlock()
+    table = SimpleTableBlock(label="Data table")
+
+    highcharts_chart_type: ClassVar[str]
+
+    # Attributes
+    show_legend = blocks.BooleanBlock(default=True, required=False)
+    show_value_labels = blocks.BooleanBlock(
+        default=False,
+        required=False,
+        help_text="For cluster charts with 3 or more series, the data labels will be hidden.",
+    )
+    use_stacked_layout = blocks.BooleanBlock(default=False, required=False)
+    show_markers = blocks.BooleanBlock(default=False, required=False)
+
+    # X axis
+    x_axis = blocks.StructBlock(
+        [
+            ("label", blocks.CharBlock(label="Label", required=False)),
+            ("min", blocks.FloatBlock(label="Minimum", required=False)),
+            ("max", blocks.FloatBlock(label="Maximum", required=False)),
+            ("tick_interval", blocks.FloatBlock(label="Tick interval", required=False)),
+        ]
+    )
+
+    # Y axis
+    y_axis = blocks.StructBlock(
+        [
+            ("label", blocks.CharBlock(required=False)),
+            ("min", blocks.FloatBlock(label="Minimum", required=False)),
+            ("max", blocks.FloatBlock(label="Maximum", required=False)),
+            ("tick_interval", blocks.FloatBlock(required=False)),
+            # TODO: implement non-stripping charblock
+            ("value_suffix", blocks.CharBlock(required=False)),
+            ("tooltip_suffix", blocks.CharBlock(required=False)),
+        ]
+    )
+
+    annotations = blocks.StreamBlock(
+        [
+            ("point", PointAnnotationBlock()),
+            # ("line", LineAnnotationBlock()),
+            # ("area", AreaAnnotationBlock()),
```

Comment:
Commented out code was left here (if left for context, let's add a comment above)

Reply:
Thanks. I have added context in da4eef0.